samples/client/petstore/rust-reqwest/src/apis/pet_api.rs

```diff
@@ -0,0 +1,349 @@
+            req_builder.header(reqwest::header::UserAgent::new(Cow::Owned(user_agent.clone())));
+        }
+
+
+        
+        if let Some(ref _apikey) = configuration.api_key {
+            req_builder.header(configuration.header_configured_api_key());
+        };
+        
+
+
+        // send request
+        let req = req_builder.build()?;
+
+        Ok(client.execute(req)?.error_for_status()?.json()?)
+    }
+
+    fn update_pet(&self, pet: ::models::Pet) -> Result<(), Error> {
+        let configuration: &configuration::Configuration = self.configuration.borrow();
+        let client = &configuration.client;
+
+
+
+        let query_string = {
+            let mut query = ::url::form_urlencoded::Serializer::new(String::new());
+
+            query.finish()
+        };
+        let uri_str = format!("{}/pet?{}", configuration.base_path, query_string);
+
+        let mut req_builder = client.put(uri_str.as_str());
+
+        if let Some(ref user_agent) = configuration.user_agent {
+            req_builder.header(reqwest::header::UserAgent::new(Cow::Owned(user_agent.clone())));
+        }
+
+
+        
+        if let Some(ref token) = configuration.oauth_access_token {
+            let auth = reqwest::header::Authorization(
+                reqwest::header::Bearer {
+                    token: token.to_owned(),
+                }
+            );
+            req_builder.header(auth.to_owned());
+        };
+
+        req_builder.json(&pet);
+
+        // send request
+        let req = req_builder.build()?;
+
+        client.execute(req)?.error_for_status()?;
+        Ok(())
+    }
+
+    fn update_pet_with_form(&self, pet_id: i64, name: &str, status: &str) -> Result<(), Error> {
+        let configuration: &configuration::Configuration = self.configuration.borrow();
+        let client = &configuration.client;
+
+
+
+        let query_string = {
+            let mut query = ::url::form_urlencoded::Serializer::new(String::new());
+
+            query.finish()
+        };
+        let uri_str = format!("{}/pet/{petId}?{}", configuration.base_path, query_string, petId=pet_id);
+
+        let mut req_builder = client.post(uri_str.as_str());
+
+        if let Some(ref user_agent) = configuration.user_agent {
+            req_builder.header(reqwest::header::UserAgent::new(Cow::Owned(user_agent.clone())));
+        }
+
+
+        
+        if let Some(ref token) = configuration.oauth_access_token {
+            let auth = reqwest::header::Authorization(
+                reqwest::header::Bearer {
+                    token: token.to_owned(),
+                }
+            );
+            req_builder.header(auth.to_owned());
+        };
+
+
+        // send request
+        let req = req_builder.build()?;
+
+        client.execute(req)?.error_for_status()?;
+        Ok(())
+    }
+
+    fn upload_file(&self, pet_id: i64, additional_metadata: &str, file: ::models::File) -> Result<::models::ApiResponse, Error> {
```

Comment:
Should we be doing something with `additional_metadata` and `file`?

Reply:
`File` management is a remaining TODO forked from `hyper` code.